### PR TITLE
Draft BLAKE2b precompile EIP

### PR DESCRIPTION
**Pull request by tjade273**

### CODE REVIEW

**Comment by Teerasa** [comment]
Free bitcoin to me.... Wonderful

**Comment by Souptacular** [comment]
EIP editor here: Please tag with editor-needs-to-review when you feel it's ready to go. It is possible that the Metropolis EIPs serve the purpose of allowing Zcash header verification in Ethereum.

@chriseth: Do the proposed metro EIPs related to zk-SNARKS make this EIP unnecessary?

Also, please enable [allow edits from maintainers](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/).

**Comment by chriseth** [comment]
@Souptacular this is unrelated to zksnarks, but zksnarks might allow for speedups here.

**Comment by Arachnid** [comment]
Please renumber as EIP-131 and update the filename and header, and we can accept this as a draft.

File: EIPS/eip-9.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+### Parameters
+
+* `METROPOLIS_FORK_BLKNUM`: TBD
+* `GBLAKEBASE`: 30
+* `GBLAKEWORD`: 6
+
+### Specification
+
+Adds a precompile at address `0x0000....0c` which accepts a variable length input interpreted as
```
Comment (axic) [inline EIPS/eip-9.md]: Cannot this be written as just `0xc`?

File: EIPS/eip-9.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    [OUTSIZE, D_1, D_2, ..., D_INSIZE]
+
+
+ where `INSIZE` is the length in bytes of the input. Throws if `OUTSIZE` is greater than 64. Returns the `OUTSIZE`-byte BLAKE2b digest, as defined in [RFC 7693](https://tools.ietf.org/html/rfc7693).
```
Comment (axic) [inline EIPS/eip-9.md]: Does `INSIZE` refer to `D_INSIZE`?

File: EIPS/eip-9.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+Gas costs would be equal to `GBLAKEBASE + GBLAKEWORD * floor(INSIZE / 32)`
+
+In order to maintain backwards compatibility, the precompile will return `0` if `CURRENT_BLOCKNUM < METROPOLIS_FORK_BLKNUM`
```
Comment (axic) [inline EIPS/eip-9.md]: What does ```return `0` ``` means? Call fails? Provided output memory is not touched?

I assume it is just a non-existent address before `METROPOLIS_FORK_BLKNUM` and therefore it s a regular failure.

**Comment by axic** [comment]
I think this should be updated with the missing fields from `eip-X.md`.

**Comment by Arachnid** [comment]
This is a courtesy notice to let you know that the format for EIPs has been modified slightly. If you want your draft merged, you will need to make some small changes to how your EIP is formatted:

 - Frontmatter is now contained between lines with only a triple dash ('---')
 - Headers in the frontmatter are now lowercase.

If your PR is editing an existing EIP rather than creating a new one, this has already been done for you, and you need only rebase your PR.

In addition, a continuous build has been setup, which will check your PR against the rules for EIP formatting automatically once you update your PR. This build ensures all required headers are present, as well as performing a number of other checks.

Please rebase your PR against the latest master, and edit your PR to use the above format for frontmatter. For convenience, here's a sample header you can copy and adapt:

```
---
eip: <num>
title: <title>
author: <author>
type: [Standards Track|Informational|Meta]
category: [Core|Networking|Interface|ERC] (for type: Standards Track only)
status: Draft
created: <date>
---
```

**Comment by axic** [comment]
Closes #152.

**Comment by Souptacular** [comment]
Please contribute to the discussion on this forum: https://ethereum-magicians.org/t/blake2b-f-precompile/3157/6

**Comment by axic** [comment]
#2024 tries to merge this PR. If that is merged, this one should be closed.

**Comment by axic** [comment]
Since the author has not responded for over a year and there are two successor EIPs (#2024, #2129), I suggest to close this.

**Comment by axic** [comment]
It seems to be that some of the original proposers converged around #2129 being the successor EIP. Closing this to avoid confusion. @tjade273 if you feel otherwise, please do not hesitate to leave a message here so we can reopen it.

**Pull request closed without merging**